src/libraries/Microsoft.PowerFx.Core/Entities/External/TableDelegationInfo.cs

```diff
@@ -0,0 +1,303 @@
+                {
+                    UngroupableProperties = new List<string>()
+                },
+                FilterFunctions = ColumnCapabilities.DefaultFilterFunctionSupport,
+                FilterSupportedFunctions = ColumnCapabilities.DefaultFilterFunctionSupport                
+            };
+        }
+    }
+
+    internal sealed class ComplexColumnCapabilities : ColumnCapabilitiesBase
+    {
+        internal Dictionary<string, ColumnCapabilitiesBase> _childColumnsCapabilities;
+
+        public ComplexColumnCapabilities()
+        {
+            _childColumnsCapabilities = new Dictionary<string, ColumnCapabilitiesBase>();
+        }
+
+        public void AddColumnCapability(string name, ColumnCapabilitiesBase capability)
+        {
+            Contracts.AssertNonEmpty(name);
+            Contracts.AssertValue(capability);
+
+            _childColumnsCapabilities.Add(name, capability);
+        }
+    }
+
+    public sealed class ColumnCapabilities : ColumnCapabilitiesBase
+    {
+        public Dictionary<string, ColumnCapabilitiesBase> Properties => _childColumnsCapabilities.Any() ? _childColumnsCapabilities : null;
+
+        private Dictionary<string, ColumnCapabilitiesBase> _childColumnsCapabilities;
+
+        public ColumnCapabilitiesDefinition Capabilities;
+
+        // Those are default CDS filter supported functions 
+        public static readonly IEnumerable<string> DefaultFilterFunctionSupport = new string[] { "eq", "ne", "gt", "ge", "lt", "le", "and", "or", "cdsin", "contains", "startswith", "endswith", "not", "null", "sum", "average", "min", "max", "count", "countdistinct", "top", "astype", "arraylookup" };
+
+        public static ColumnCapabilities DefaultColumnCapabilities => new ColumnCapabilities()
+        {
+            Capabilities = new ColumnCapabilitiesDefinition()
+            {
+                FilterFunctions = DefaultFilterFunctionSupport,
+                QueryAlias = null,
+                IsChoice = null
+            },
+            _childColumnsCapabilities = new Dictionary<string, ColumnCapabilitiesBase>()
+        };
+
+        private ColumnCapabilities()
+        {
+        }
+
+        public void AddColumnCapability(string name, ColumnCapabilitiesBase capability)
+        {
+            Contracts.AssertNonEmpty(name);
+            Contracts.AssertValue(capability);
+
+            _childColumnsCapabilities.Add(name, capability);
+        }
+
+        public ColumnCapabilities(ColumnCapabilitiesDefinition capability)
+        {
+            Contracts.AssertValueOrNull(capability);
+
+            Capabilities = capability;
+            _childColumnsCapabilities = new Dictionary<string, ColumnCapabilitiesBase>();
+        }
+    }
+
+    public sealed class ColumnCapabilitiesDefinition
+    {        
+        public IEnumerable<string> FilterFunctions
+        {
+            get => _filterFunctions ?? DefaultFilterFunctionSupport;
+            set => _filterFunctions = value;
```

Comment:
>value [](http://example.com/codeflow?start=38&length=5)

Why do we need a public setter for this? We may not be able to support all filter operations in powerFx. It should be known set.

Reply:
init => ... is better, I agree